test_app/tests/test_models.py

```diff
@@ -36,3 +36,23 @@ def test_parse_with_invalid_data_type(self):
         with self.assertRaises(ParseException) as ex:
             basic_class = BasicClass.parse(1)
         self.assertEqual(ex.exception.args[0], ErrorMessages.NO_DATA_OR_DATA_NOT_DICT)
+
+
+class BasicClassChangeTrackingFieldsTests(unittest.TestCase):
+
+    def test_change_tracking_fields_without_changes(self):
+        # Arrange
+        basic_class = BasicClass(name='Initial Value')
+        # Act
+        # Assert
+        self.assertFalse(basic_class.change_tracking_field_has_changed('name'))
+        self.assertEqual(basic_class._name_initial, 'Initial Value')
+
+    def test_change_tracking_fields_with_changes(self):
+        # Arrange
+        basic_class = BasicClass(name='Initial Value')
```

Comment:
I would maybe actually save this to the db and re-query. So it can be initialized from the database rather than in memory